ceph-releases/kraken/ubuntu/16.04/daemon/start_mon.sh

```diff
@@ -1,6 +1,74 @@
 #!/bin/bash
 set -e
 
+IPV4_REGEXP='[0-9]\{1,3\}\.[0-9]\{1,3\}\.[0-9]\{1,3\}\.[0-9]\{1,3\}'
+IPV4_NETWORK_REGEXP="$IPV4_REGEXP/[0-9]\{1,2\}"
+
+function flat_to_ipv6 {
+  # Get a flat input like fe800000000000000042acfffe110003 and output fe80::0042:acff:fe11:0003
+  # This input usually comes from the ipv6_route or if_inet6 files from /proc
+
+  # First, split the string in set of 4 bytes with ":" as separator
+  value=$(echo "$@" | sed -e 's/.\{4\}/&:/g' -e '$s/\:$//')
+
+  # Let's remove the useless 0000 and "::"
+  value=${value//0000/:};
+  while $(echo $value | grep -q ":::"); do
+    value=${value//::/:};
+  done
+  echo $value
+}
+
+function get_ip {
+  NIC=$1
+  # IPv4 is the default unless we specify it
+  IP_VERSION=${2:-4}
+  # We should avoid reporting any IPV6 "scope local" interface that would make the ceph bind() call to fail
+  if command -v ip &>/dev/null; then
+    ip -$IP_VERSION -o a s $NIC | grep "scope global" | awk '{ sub ("/..", "", $4); print $4 }' || true
```

Comment:
we are running under bash -e meaning a faulty commands makes the whole aborting.
As we are trying to detect something, I don't want to fail here because of a mis-dectection.
That's partciulary true with IPv6 since we try to detect something that is very unlikely to be present.

We have some code checking lately if MON_IP is empty and fail at that point with a clear message.

So I'm putting here a || true to let that code triggering the real exception.